src/Resolver/AggregateResolver.php

```diff
@@ -72,77 +39,27 @@ public function count()
      *
      * @return Traversable<int, ResolverInterface>
      */
-    #[ReturnTypeWillChange]
-    public function getIterator()
+    public function getIterator(): Traversable
     {
         return $this->queue;
     }
 
-    /**
-     * Attach a resolver
-     *
-     * @param  int $priority
-     * @return $this
-     */
-    public function attach(Resolver $resolver, $priority = 1)
+    public function attach(ResolverInterface $resolver, int $priority = 1): self
     {
         $this->queue->insert($resolver, $priority);
         return $this;
     }
 
-    /**
-     * Resolve a template/pattern name to a resource the renderer can consume
-     *
-     * @param  string $name
-     * @return false|string
-     */
-    public function resolve($name, ?Renderer $renderer = null)
+    public function resolve(string $name): string
     {
-        $this->lastLookupFailure      = false;
-        $this->lastSuccessfulResolver = null;
-
-        if (0 === count($this->queue)) {
-            $this->lastLookupFailure = static::FAILURE_NO_RESOLVERS;
-            return false;
-        }
-
         foreach ($this->queue as $resolver) {
-            /**
-             * @todo This loop should be modified to try { return resolve } catch { continue } in v3.0
-             */
-            $resource = $resolver->resolve($name, $renderer);
-            if (is_string($resource)) {
-                // Resource found; return it
-                $this->lastSuccessfulResolver = $resolver;
-                return $resource;
+            try {
+                return $resolver->resolve($name);
+            } catch (TemplateCannotBeFound) {
+                continue;
```

Comment:
Note: using exceptions for this can be really expensive, and this is critical path for most apps.

Reply:
👍 I've added, used and tested `ResolverInterface::has($name): bool`